sys-cluster/charliecloud: Version bumps to 0.17, 0.18.

olifre opened the pull request:
We now use a newly added `configure` flag to remove `-Werror` instead of patching it out. 
No python 3.8 yet due to https://bugs.gentoo.org/723490 .

olifre commented:
Since upstream made a new release today with several fixes, I updated this pending PR with a second version bump.

juippis commented on sys-cluster/charliecloud/charliecloud-0.18.ebuild:
```diff
@@ -0,0 +1,79 @@
+
+PYTHON_COMPAT=( python3_{6,7} )
+
+inherit autotools eutils python-single-r1
```
This has grown a bit outdated and there's an eclass called `optfeature` now to use instead of `eutils`.

olifre replied:
Thanks! 
Is there a mailing list to stay informed of new `eclass` such as this one (I missed that on `gentoo-dev-announce`)?

thesamesam replied:
I think it's just gentoo-dev. It's not the worst idea that final eclass changes could be summarised to gentoo-dev-announce though.

olifre replied:
Ah, ok! Indeed, I am only on `gentoo-dev-announce` since as a proxy-maintainer, the full`gentoo-dev` list is a bit much (would be fine as RSS feed or such, but not for my mailbox which is currently eaten by other projects). If a summary of new eclasses / change of recommendations could be copied to `gentoo-dev-announce`, I believe many would appreciate this :-).

thesamesam replied:
I'm wondering if a mail filter could work for now, at least? This is what I do to separate mailing lists into their own folders.

juippis reviewed (commented):
Also if possible drop the commit for 0.17, since it'd just be noise in the git logs of main tree. I can do it for you too while merging.

olifre commented:
> Also if possible drop the commit for 0.17, since it'd just be noise in the git logs of main tree. I can do it for you too while merging.

Done, I squashed adding 0.17 and 0.18 in one commit (I think that is what you meant, correct?). 
I also rebased on current HEAD, since in the meantime, `optfeature` was already added to `inherit` by a global change for the old version I dropped as part of this PR (causing a conflict).

thesamesam commented:
I think the idea was "Do we really want both N and N-1 (where N is the latest version)?"

Usually there's not much purpose to this. Is there a reason you want both in tree, not just the latest?

juippis commented:
Yep, since 0.17 and 0.18 will have same KEYWORDS and they're introduced in the same time, no one will install 0.17.

olifre commented:
> Usually there's not much purpose to this. Is there a reason you want both in tree, not just the latest?

Indeed, that's a good point. 

> Yep, since 0.17 and 0.18 will have same KEYWORDS and they're introduced in the same time, no one will install 0.17.

I've dropped 0.17 from history now, so there's only the version bump to 0.18 and dropping a very old release now in this PR ;-).
